src/services/llms/base.py

```diff
@@ -1,13 +1,79 @@
+# src/services/llm/base.py
 from abc import ABC, abstractmethod
-from typing import List, Dict, Any
+import json
+from typing import List, Dict
+from unidiff import Hunk, PatchedFile
 from ...core.models import PRDetails
 
-
 class BaseLLMService(ABC):
+    """
+    Abstract base class for LLM services defining the common interface
+    that all LLM implementations must follow.
+    """
+    
     @abstractmethod
-    def create_prompt(self, file_path: str, diff_content: str, pr_details: PRDetails) -> str:
+    def create_prompt(self, file: PatchedFile, hunk: Hunk, pr_details: PRDetails) -> str:
+        """
+        Create a prompt for the LLM model.
+        
+        Args:
+            file: The file being reviewed
+            hunk: The code hunk to review
+            pr_details: Pull request details
+            
+        Returns:
+            str: Formatted prompt for the LLM
```

Comment:
The method name `get_ai_response` is not descriptive.  Consider renaming it to something like `get_llm_response` or `retrieve_llm_response` to better communicate that it's retrieving a response from a large language model.  The improved name will also enhance readability and understanding.